File: LeetCode/Easy/1518-water-bottles/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        
        total_bottles = numBottles
        remainder = 0
        while numBottles:
            quotient, remainder = divmod(numBottles + remainder, numExchange)
            total_bottles += quotient
            numBottles = quotient

        return total_bottles            
        

File: LeetCode/Easy/1518-water-bottles/1518-water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {

        int maximumBottles = numBottles;
        int remainder = 0;
        int quotient = 0;
        while(numBottles > 0){
            quotient = (numBottles + remainder) / numExchange;
            remainder = (numBottles + remainder) % numExchange;
            maximumBottles += quotient;
            numBottles = quotient;
        }
        return maximumBottles;
    }
};